### ISSUE TITLE
JIT Script module support for pad_packed_sequence and pack_padded_sequence

### ISSUE THREAD

**Issue by houseroad** [pytorch/pytorch#16664]
## 🚀 Feature
Users would like to implement LSTM module as jit script module. However, `pack_padded_sequence` and `pad_packed_sequence` does not support it yet.

The code to reproduce the problem:

```python
class Model(torch.jit.ScriptModule):
    def __init__(self):
        super().__init__()
        self.embedding = nn.Embedding(VOCAB_SIZE, EMBEDDING_SIZE)
        self.lstm = nn.LSTM(EMBEDDING_SIZE, LSTM_SIZE, num_layers=1, bidirectional=True)

    @jit.script_method
    def forward(self, tokens, seq_lengths):
        embedded = self.embedding(tokens)
        rnn_input = pack_padded_sequence(
            embedded,
            seq_lengths.int(), True, True,
        )
        rep, unused_state = self.lstm(rnn_input)
        unpacked, _ = pad_packed_sequence(
            rep,
            batch_first=True,
            padding_value=0.0,
            total_length=embedded.size(1),
        )
        return unpacked
```

cc: @jamesr66a @pritamdamania

**Comment by julioasotodv** [comment]
@houseroad just to confirm: you get an error similar to this, right?

```python
for operator (Tensor 0, Tensor 1, Tensor 2, Tensor 3) -> Tensor:
expected a value of type Tensor for argument '2' but found bool

rnn_input = pack_padded_sequence(
            embedded,
            seq_lengths.int(), 
           True,
           ~~~~~~ <--- HERE
           True)
```

**Comment by driazati** [comment]
Fixed in #23249
